homeassistant/components/slide/__init__.py

```diff
@@ -0,0 +1,157 @@
+            slidenew = hass.data[DOMAIN][SLIDES].setdefault(uid, {})
+            slidenew["mac"] = uid
+            slidenew["id"] = slide["id"]
+            slidenew["name"] = slide["device_name"]
+            slidenew["state"] = None
+            oldpos = slidenew.get("pos")
+            slidenew["pos"] = None
+            slidenew["online"] = False
+
+            if "device_info" not in slide:
+                _LOGGER.error(
+                    "Slide %s (%s) has no device_info Entry=%s",
+                    slide["id"],
+                    slidenew["mac"],
+                    slide,
+                )
+                continue
+
+            # Check if we have pos (OK) or code (NOK)
+            if "pos" in slide["device_info"]:
+                slidenew["online"] = True
+                slidenew["pos"] = slide["device_info"]["pos"]
+                slidenew["pos"] = max(0, min(1, slidenew["pos"]))
+
+                if oldpos is None or oldpos == slidenew["pos"]:
+                    slidenew["state"] = (
+                        STATE_CLOSED if slidenew["pos"] > 0.95 else STATE_OPEN
+                    )
+                elif oldpos < slidenew["pos"]:
+                    slidenew["state"] = (
+                        STATE_CLOSED if slidenew["pos"] >= 0.95 else STATE_CLOSING
+                    )
+                else:
+                    slidenew["state"] = (
+                        STATE_OPEN if slidenew["pos"] <= 0.05 else STATE_OPENING
+                    )
+            elif "code" in slide["device_info"]:
+                _LOGGER.warning(
+                    "Slide %s (%s) is offline with " "code=%s",
+                    slide["id"],
+                    slidenew["mac"],
+                    slide["device_info"]["code"],
+                )
+            else:
+                _LOGGER.error(
+                    "Slide %s (%s) has invalid device_info %s",
+                    slide["id"],
+                    slidenew["mac"],
+                    slide["device_info"],
+                )
+
+            _LOGGER.debug("Updated entry=%s", slidenew)
+
+    async def retry_setup(now):
+        """Retry setup if a connection/timeout happens on Slide API."""
+        await async_setup(hass, config)
+
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][SLIDES] = {}
+
+    username = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+    scaninterval = config[DOMAIN][CONF_SCAN_INTERVAL]
+
+    hass.data[DOMAIN][API] = GoSlideCloud(username, password)
+
+    try:
+        result = await hass.data[DOMAIN][API].login()
+    except (goslideapi.ClientConnectionError, goslideapi.ClientTimeoutError) as err:
+        _LOGGER.error(
+            "Error connecting to Slide Cloud: %s, going to retry in %s seconds",
+            str(err),
+            DEFAULT_RETRY,
+        )
+        async_call_later(hass, DEFAULT_RETRY, retry_setup)
+        return True
+
+    if result:
```

Comment:
If we invert this check and log error and return False if true, we can outdent the debug log.
```suggestion
    if not result:
```